lang/javascript.go

```diff
@@ -0,0 +1,40 @@
+package lang
+
+import (
+	"github.com/safedep/code/core"
+	sitter "github.com/smacker/go-tree-sitter"
+	"github.com/smacker/go-tree-sitter/javascript"
+)
+
+const javascriptLanguageName = "javascript"
+
+type javascriptLanguage struct{}
+
+var _ core.Language = (*javascriptLanguage)(nil)
+
+func NewJavascriptLanguage() (*javascriptLanguage, error) {
+	return &javascriptLanguage{}, nil
+}
+
+func (l *javascriptLanguage) Name() string {
+	return javascriptLanguageName
+}
+
+func (l *javascriptLanguage) Meta() core.LanguageMeta {
+	return core.LanguageMeta{
+		Name:                 javascriptLanguageName,
+		Code:                 core.LanguageCodeJavascript,
+		ObjectOriented:       true,
```

Comment:
I think this will be false. Javascript is prototype language and not object oriented AFAIK

Reply:
wow! Lets go ahead with OOP then now that we know better :)